Zend/zend_builtin_functions.c

```diff
@@ -491,12 +491,16 @@ ZEND_FUNCTION(func_num_args)
 {
 	zend_execute_data *ex = EX(prev_execute_data);
 
-	if (!(ZEND_CALL_INFO(ex) & ZEND_CALL_CODE)) {
-		RETURN_LONG(ZEND_CALL_NUM_ARGS(ex));
-	} else {
+	if (ZEND_CALL_INFO(ex) & ZEND_CALL_CODE) {
 		zend_error(E_WARNING, "func_num_args():  Called from the global scope - no function context");
 		RETURN_LONG(-1);
 	}
+
+	if (zend_forbid_dynamic_call("func_num_args()") == FAILURE) {
```

Comment:
func_num_args() is not a problem from a safety perspective, but the current behavior for dynamic func_num_args() calls is inconsistent depending on whether we're in namespaced / non-namespaced code: https://3v4l.org/inP5v -- anything that inspects the "next frame up" will have this issue, this includes all func_* functions.

Furthermore, while func_num_args() is not an issue for the current optimizer, it would become a problem once we implement inlining (in which case func_num_args() would get the argument number of the function we inlined into).

As such I think we should forbid it as well.